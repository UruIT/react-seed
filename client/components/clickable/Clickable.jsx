import React from 'react';
import PropTypes from 'prop-types';
import styles from './clickable.scss';

const dialog = e => alert(e.target.textContent);

const Clickable = ({ text, onClick }) => (
	<div className={styles.container} onClick={onClick}>
		{ text }
	</div>
);

Clickable.propTypes = {
	onClick: PropTypes.func,
	text: PropTypes.string
};

Clickable.defaultProps = {
	onClick: dialog,
	text: '...'
};

export default Clickable;
