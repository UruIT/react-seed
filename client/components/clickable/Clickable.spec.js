import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Clickable from './Clickable';

describe('<Clickable/>', () => {
	it('snapshot - default props', () => {
		const tree = renderer.create(<Clickable />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('snapshot - with TEXT', () => {
		const tree = renderer.create(<Clickable content="TEXT" />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe('Clickable - on Click', () => {
	it('should call onClick callback', () => {
		const onClickCb = jest.fn();

		const app = shallow(<Clickable onClick={onClickCb} />);

		app.simulate('click');

		expect(onClickCb).toBeCalled();
	});
});
