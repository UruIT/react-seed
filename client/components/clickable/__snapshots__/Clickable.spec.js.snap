// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clickable/> snapshot - default props 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  ...
</div>
`;

exports[`<Clickable/> snapshot - with TEXT 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  TEXT
</div>
`;
